portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -15,18 +15,33 @@
 package com.google.sps.servlets;
 
 import java.io.IOException;
+import java.util.ArrayList;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+@WebServlet("/comment")
 public class DataServlet extends HttpServlet {
+  private ArrayList<String> commentsString = new ArrayList<String>();
```

Comment:
Are we going to commentsString  to a different instance of an ArrayList?
Does any user of commentsString  need to care about the specific selection of List?

Reply:
Any progress on this?

Reply:
No, other users won't need to care about this. WIll change to an immutable string.

Reply:
Sorry, what do you mean by "Are we going to commentsString to a different instance of an ArrayList?"

Reply:
I think Robert means that if we know that commentsString is always going to point to the same instance of ArrayList (i.e. we don't assign a different ArrayList to it later on), then we could [make the variable final](https://stackoverflow.com/questions/13079365/declaring-a-list-field-with-the-final-keyword) (because it won't change).

For the second question, I think he's referring to preferring to use the interface as the variable type, e.g.
`private List<String> commentsString = new ArrayList<String>();` instead of `private ArrayList<String> commentsString = new ArrayList<String>();`.

Feel free to correct me Robert!

Reply:
I see, thanks! I'll quickly change it now.

Reply:
You can read my mind. I am sorry as it was not sufficiently clear.

Reply:
Combined, I would suggest:
private final List<String> commentsString = new ArrayList<>();

Reply:
Yep changed!